### PR TITLE
Update build-images and integration tests

### PR DESCRIPTION
**Pull request by corneil**
Add JDK 21 and remove 8, 11 from build-images.
Remove JDK pre 17 from container tests.
Add JDK 21 to container tests.
Update DataflowOAuthIT to dump last error output from curl.

### CODE REVIEW

File: spring-cloud-dataflow-server/src/test/java/org/springframework/cloud/dataflow/integration/test/db/AbstractDatabaseTests.java [hunk 1]
```diff
@@ -62,35 +62,18 @@ public void testLatestSharedDb() {
...
-	public void testLatestSharedDbJdk8() {
+	public void testLatestSharedDbJdk21() {
 		log.info("Running testLatestSharedDb()");
 		// start defined database
 		this.dataflowCluster.startSkipperDatabase(getDatabaseTag());
 		this.dataflowCluster.startDataflowDatabase(getDatabaseTag());
 
 		// start defined skipper server and check it started
-		this.dataflowCluster.startSkipper(TagNames.SKIPPER_main + "-jdk8");
+		this.dataflowCluster.startSkipper(TagNames.SKIPPER_main + "-jdk21");
```
Comment (cppwfs) [inline spring-cloud-dataflow-server/src/test/java/org/springframework/cloud/dataflow/integration/test/db/AbstractDatabaseTests.java:72]: Should we add a test for 17 as well as 21?

**Review by cppwfs — CHANGES_REQUESTED** [review]

File: spring-cloud-dataflow-server/src/test/java/org/springframework/cloud/dataflow/integration/test/db/AbstractDatabaseTests.java (hunk 1 above)
Comment (corneil) [reply spring-cloud-dataflow-server/src/test/java/org/springframework/cloud/dataflow/integration/test/db/AbstractDatabaseTests.java:72]: We had tests for 8, 11, 17 so in the same vein I thought we should have 17 and 21.

Comment (cppwfs) [reply spring-cloud-dataflow-server/src/test/java/org/springframework/cloud/dataflow/integration/test/db/AbstractDatabaseTests.java:72]: I stand corrected.   I didn't see that we had 17 already.

**Comment by cppwfs** [comment]
Kicked the tires and noticed that a couple tests fail.   
We do need to add Java 21 to the container list here: https://github.com/spring-cloud/spring-cloud-dataflow/blob/main-3/spring-cloud-dataflow-server/src/test/java/org/springframework/cloud/dataflow/integration/test/db/AbstractDataflowTests.java#L159-L166.  Else we have tests that will fail.

File: spring-cloud-dataflow-server/src/test/java/org/springframework/cloud/dataflow/integration/test/db/AbstractDataflowTests.java [hunk 2]
```diff
@@ -169,7 +169,7 @@ protected List<ClusterContainer> getSkipperContainers() {
 	protected List<ClusterContainer> getDataflowContainers() {
 		ArrayList<ClusterContainer> containers = new ArrayList<>(DATAFLOW_CONTAINERS);
 		containers.add(ClusterContainer.from(TagNames.DATAFLOW_main, DATAFLOW_IMAGE_PREFIX + getDataflowLatestVersion()));
-		List<Integer> jdkTags = Arrays.asList(8, 11, 17);
+		List<Integer> jdkTags = Arrays.asList(17, 21);
```
Comment (cppwfs) [inline spring-cloud-dataflow-server/src/test/java/org/springframework/cloud/dataflow/integration/test/db/AbstractDataflowTests.java:172]: Kicked the tires and noticed that a couple tests fail.
We do need to add Java 21 to the container list here: https://github.com/spring-cloud/spring-cloud-dataflow/blob/main-3/spring-cloud-dataflow-server/src/test/java/org/springframework/cloud/dataflow/integration/test/db/AbstractDataflowTests.java#L159-L166. Else we have tests that will fail.

**Review by cppwfs — CHANGES_REQUESTED** [review]

File: spring-cloud-dataflow-server/src/test/java/org/springframework/cloud/dataflow/integration/test/db/AbstractDataflowTests.java (hunk 2 above)
Comment (corneil) [reply spring-cloud-dataflow-server/src/test/java/org/springframework/cloud/dataflow/integration/test/db/AbstractDataflowTests.java:172]: Fixed

**Review by cppwfs — APPROVED** [review]
LGTM

Thank you for the updates!!!

**Comment by cppwfs** [comment]
LGTM Squashed Merged

Thank you @corneil

**Pull request closed without merging**